### DISCUSSION TITLE
working title until I think of something more usefull

### DISCUSSION THREAD

**Discussion by chris-kruining** [aleksilassila/reiverr#126] [General]
Hey,

Love the project, you are basically making the thing I was about to make myself. So I am totally onboard for it.

I see plenty of help wanted tags in the issues, is this still the case? I'd love to throw myself at the project and implement a bunch of features. I mean if I was going to do it anyway, might as well do it here :P

I also have some feature suggestions:
- delcaritive configuration (perhaps already available via a docker volume???)
- support for lg webos (probably pretty niche, but it is what I have at home)
- not so much a feature request as a ux request: the goal I had for my own project was "netflix but with my own backend"
  is this something you also have in mind?

I did read the homepage and I am wondering if I am maybe just a bit early. But I am also fine with you getting me up to speed and me helping out with finishing the migration.

Let me know! looking forward to the possibilities!

**Comment by aleksilassila** [comment]
Hey, glad to hear you like the project! I'm currently just about to make the transition to the 2.0 version, with the release of the first alpha. It's a complete rewrite of the project, that introduces a dedicated backend written in NestJS and dropping sveltekit in favor of plain svelte.

That being said, I'll try to go through and clean away most of the issues in the old taskboard, and create a new roadmap for the 2.0 version (this is because I don't plan on updating the old v0.x codebase anymore, once the switch has been made). So I'd recommend you check back next week, when I've hopefully got all the ducks in a row regarding merging everything and defining a new roadmap.

About the suggestions:
> delcaritive configuration (perhaps already available via a docker volume???)

You mean defining the credentials with environment variables? 2.0 will complicate this a bit, since I'll be adding support for multiple users, but I guess it would be possible (and even useful / good practice) to define let's say the admin account with environment variables. Right now in 2.0 anyone can create the first account without authentication, and that account will be made an admin account. 

> support for lg webos (probably pretty niche, but it is what I have at home)

This is absolutely doable, but I'd need your help on this. The main reason why the project was rewritten was to make it a SPA that can primarily target TVs, so everything including dpad navigation should work out of the box on TVs now. Only thing is that I only own a samsung tv, so if you could figure out how to package and install the app on WebOS and then maybe do a PR, that would be awesome.

>not so much a feature request as a ux request: the goal I had for my own project was "netflix but with my own backend"
is this something you also have in mind?

This is absolutely what I'm aiming for, and with the dedicated backend server we will get closer there. I've also done quite a lot of background work and research on how one would even connect multiple servers / content libraries together via reiverr to create a mesh network where everyone can share their libraries with each other. It's an interesting and complex concept, so I'll probably make a separate thread for discussing it later. Anyway, the support for multiple users will get this a lot closer to the Netflix experience, if that's what youre thinking. It will allow for separate recommendations and even content libraries if necessary.

Cheers!

**Reply by GrabbenD** [reply]
> if you could figure out how to package and install the app on WebOS

It's your lucky day guys 🙂 

I spent hours fighting LG's WebOS 2 (TV UF950V) and after a lot of trail and error managed to figure it out. Made a mini guide here:

https://stackoverflow.com/a/77513090/16780116

---

> SPA that can primarily target TVs

I tried a lot of high-end expensive TVs from LG, Samsung, Philips & Sony and I'm NEVER AGAIN wasting time on using TV's operating system. App compatibility, software limitations, problems with codecs (due to proprietary + outdated components in the OS) and slowness are just too common

I firmly [believe](https://github.com/aleksilassila/reiverr/issues/98#issuecomment-1861254590) that focusing on a rock solid Android TV implementation will result in the best user experience and least friction for developers. Tried 4 different Android TV devices and every single one is miles ahead of TVs (and you can even compile your own WebView to add missing codecs or [performance enhancements](https://github.com/Alex313031/Thorium-Android/releases))!

I applaud your dedication @aleksilassila 
Hope 2.0 goes well

**Reply by aleksilassila** [reply]
Heh, I have similar feelings towards Tizen, gave up so many times. Good to hear that maybe Android TV won't be as painful to work with.

**Reply by chris-kruining** [reply]
> You mean defining the credentials with environment variables? 2.0 will complicate this a bit, since I'll be adding support for multiple users, but I guess it would be possible (and even useful / good practice) to define let's say the admin account with environment variables. Right now in 2.0 anyone can create the first account without authentication, and that account will be made an admin account.

not environment variables, a config file, a good old json, or yaml, or toml, or whatever floats your boat.

**Comment by chris-kruining** [comment]
I wouldn't mind giving user management a shot, I have experience with OIDC, and the *arr apps seem to have LDAP (never worked with it, but I'd like to learn) support. Hopefully that way reiverr doesn't need to roll, implement, and maintain its own version. and I don't have to manage users in 10 places :P

**Comment by chris-kruining** [comment]
I haven't forgotten about this, I've just been really busy. hopefully next week I can find some time
